python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -244,13 +372,25 @@ def getBackgroundReferenceScaler(self, dataRef):
         return refImageScaler
 
     def prepareInputs(self, refList):
-        """Prepare the input warps for coaddition
-
-        This involves measuring weights and constructing image scalers
-        for each of the inputs.
-
-        @param refList: List of data references to tempExp
-        @return Struct:
+        """!
+        \brief Prepare the input warps for coaddition by measuring the weight for each warp and the scaling 
+        for the photometric zero point.
+
+        Each coaddTempExp has its own photometric zeropoint and background variance. Before coadding these 
+        coaddTempExps together, compute a scale factor to normalize the photometric zeropoint and the weight 
+        for each coaddTempExp. 
+
+        For each coaddTempExp, use the \ref ScaleZeroPointTask_ "scaleZeroPoint" subtask to compute a 
+        imageScaler object that can be used to scale the coaddTempExp photometric zeropoint to that of the 
+        reference coaddTempExp. Extract and scale the maskedImage in the coaddTempExp to the common zeropoint 
+        using \ref imageScaler.scaleMaskedImage "scaleMaskedImage".
+        Define a \ref afwMath.StatisticsControl "StatisticsControl" object to compute the mean of the variance
+        plane of the masked image.
+        Assign weight equal to inverse of the mean of the variance plane (i.e. weight noisy coaddTempExps
+        less). Return the weights and imageScaler objects.
```

Comment:
This is a similar comment as before and applies to many functions below.  I think you can condense this and leave out details such as which objects are constructed and document the functionality of the code instead of the implementation.  I think the details make it more difficult to understand what is happening.